### PR TITLE
fix: add check for empty array parameters in batch `ERC725X.execute(uint256[],address[],uint256[],bytes[])` / `ERC725Y.setData(bytes32[],bytes[])`

### PR DESCRIPTION
**Pull request by CJ42**
# What does this PR introduce?

## 🐛 Bug

When the batch functions below are called with empty array parameters:
- `ERC725X.execute(uint256[],address[],uint256[],bytes[])`
- `ERC725Y.setData(bytes32[],bytes[])`

Nothing happen and the contract call return successfully and silently. Reason being the internal function never enter inside the `for` loops.

Add extra checks to test for empty array parameters in these batch functions.

## 🧪 Tests

Add tests when:
- all the array parameters are empty arrays.
- at least one of the array parameters is an empty array.

### PR Checklist

- [x] Wrote Tests
- [x] Wrote Documentation
- [x] Ran `npm run lint`
- [x] Ran `npm run build`
- [x] Ran `npm run test`

### CODE REVIEW

**Review by b00ste — APPROVED** [review]

File: implementations/contracts/ERC725XCore.sol [hunk 1]
```diff
@@ -130,6 +130,13 @@ abstract contract ERC725XCore is OwnableUnset, ERC165, IERC725X {
         uint256[] memory values,
         bytes[] memory datas
     ) internal virtual returns (bytes[] memory) {
+        if (
```
Comment (YamenMerhi) [inline implementations/contracts/ERC725XCore.sol:133]: Maybe we can move this check below the check of `ERC725X_ExecuteParametersLengthMismatch` and then we can check just one parameter that is equal to 0.

File: implementations/contracts/ERC725YCore.sol [hunk 2]
```diff
@@ -72,6 +72,10 @@ abstract contract ERC725YCore is OwnableUnset, ERC165, IERC725Y {
         /// @dev do not allow to send value by default when setting data in ERC725Y
         if (msg.value != 0) revert ERC725Y_MsgValueDisallowed();
 
+        if (dataKeys.length == 0 || dataValues.length == 0) {
```
Comment (YamenMerhi) [inline implementations/contracts/ERC725YCore.sol:75]: Same here

**Review by YamenMerhi — CHANGES_REQUESTED** [review]
Nice one!

File: implementations/contracts/ERC725XCore.sol (hunk 1 above)
Comment (CJ42) [reply implementations/contracts/ERC725XCore.sol:133]: love your suggested comment! 🎉

**Review by YamenMerhi — APPROVED** [review]
